mesonbuild/dependencies/dev.py

```diff
@@ -334,7 +334,7 @@ def _set_new_link_args(self, environment: 'Environment') -> None:
 
         link_args = ['--link-static', '--system-libs'] if self.static else ['--link-shared']
         self.link_args = self.get_config_value(
-            ['--libs', '--ldflags'] + link_args + list(self.required_modules),
+            ['--libs', '--ldflags'] + link_args + sorted(self.required_modules),
```

Comment:
It's probably easier to just store the modules using mesonlib.OrderedSet instead of pain sets, then we won't run into other issues like this elsewhere

Reply:
Thank you for the suggestion! That sounds like a much better suggestion, since mesonlib includes an OrderedSet.